flink-filesystems/flink-s3-fs-base/src/test/java/org/apache/flink/fs/s3/common/HAJobRunOnMinioS3StoreITCase.java

```diff
@@ -0,0 +1,123 @@
+import org.apache.flink.runtime.highavailability.AbstractHaJobRunITCase;
+import org.apache.flink.runtime.highavailability.JobResultStoreOptions;
+import org.apache.flink.runtime.testutils.CommonTestUtils;
+
+import org.apache.flink.shaded.guava30.com.google.common.collect.Iterables;
+
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.time.Duration;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * {@code HAJobRunOnMinioS3StoreITCase} covers a job run where the HA data is stored in Minio. The
+ * implementation verifies whether the {@code JobResult} was written into the FileSystem-backed
+ * {@code JobResultStore}.
+ */
+public abstract class HAJobRunOnMinioS3StoreITCase extends AbstractHaJobRunITCase {
+
+    private static final String CLUSTER_ID = "test-cluster";
+    private static final String JOB_RESULT_STORE_FOLDER = "jrs";
+
+    @RegisterExtension
+    private static final AllCallbackWrapper<TestContainerExtension<MinioTestContainer>>
+            MINIO_EXTENSION =
+                    new AllCallbackWrapper<>(new TestContainerExtension<>(MinioTestContainer::new));
+
+    private static MinioTestContainer getMinioContainer() {
+        return MINIO_EXTENSION.getCustomExtension().getTestContainer();
+    }
+
+    private static String createS3URIWithSubPath(String... subfolders) {
+        return getMinioContainer().getS3UriForDefaultBucket() + createSubPath(subfolders);
+    }
+
+    private static List<S3ObjectSummary> getObjectsFromJobResultStore() {
+        return getMinioContainer()
+                .getClient()
+                .listObjects(
+                        getMinioContainer().getDefaultBucketName(),
+                        createSubPath(CLUSTER_ID, JOB_RESULT_STORE_FOLDER))
+                .getObjectSummaries();
+    }
+
+    private static String createSubPath(String... subfolders) {
+        final String pathSeparator = "/";
+        return pathSeparator + StringUtils.join(subfolders, pathSeparator);
+    }
+
+    @Override
+    protected String getHAStoragePath() {
+        return createS3URIWithSubPath(CLUSTER_ID);
+    }
+
+    @Override
+    protected Configuration createConfiguration() {
+        final Configuration config = new Configuration();
+
+        getMinioContainer().setS3ConfigOptions(config);
+
+        // JobResultStore configuration
+        config.set(JobResultStoreOptions.DELETE_ON_COMMIT, Boolean.FALSE);
+        config.set(
+                JobResultStoreOptions.STORAGE_PATH,
+                createS3URIWithSubPath(CLUSTER_ID, JOB_RESULT_STORE_FOLDER));
+
+        return config;
+    }
+
+    @Override
+    protected void runAfterJobTermination() throws Exception {
+        final String jsonExtension = ".json";
+        final String dirtyJsonExtesion = "_DIRTY" + jsonExtension;
```

Comment:
Can we link this somehow to the JRS implementation? The FS module shouldn't have to know how the file is named etc.